Source/Redemption/Private/GargoyleController.cpp

```diff
@@ -15,4 +20,32 @@ AGargoyleController::AGargoyleController()
 	SightSenseConfig->LoseSightRadius = SIGHT_RADIUS;
 
 	PerceptionComp->ConfigureSense(*SightSenseConfig);
+}
+
+void AGargoyleController::BeginPlay()
+{
+	Super::BeginPlay();
+
+	if (AGargoyle* Gargoyle = GET_GARGOYLE)
+	{
+		SetStandingIndex(Gargoyle);
+	}
+}
+
+void AGargoyleController::SetStandingIndex(AGargoyle* Gargoyle)
+{
+	TArray<APatrolPathIndicator*> StadingPoints = Gargoyle->GetStandingPoints();
+	float ClosestDistance = 10000000;
+	APatrolPathIndicator* CurrentStandingPos = nullptr;
+
+	for (APatrolPathIndicator* Current : StadingPoints)
+	{
+		if (FVector::Dist(Gargoyle->GetActorLocation(), Current->GetActorLocation()) < ClosestDistance)
+		{
+			ClosestDistance = FVector::Dist(Gargoyle->GetActorLocation(), Current->GetActorLocation());
+			CurrentStandingPos = Current;
+		}
+	}
+
+	GetBlackboardComponent()->SetValueAsInt("LocationIndex", Gargoyle->GetStandingIndex(CurrentStandingPos));
```

Comment:
shouldn't be possible, but should you wrap this line in a CurrentStandingPos != nullptr in case it loops and has none?

Reply:
GetStandingIndex returns -1 if the argument is not part of the list, if it is nullptr for example, so the AI would not break. I think there is no need to